### ISSUE TITLE
Tunneling rtmp

### ISSUE THREAD

**Issue by mananchawla2005** [cloudflare/cloudflared#497]
Hi, I used cloudflare tunnels and it seemed to be fun. I was wondering if we could tunnel rtmp traffic as well? I searched for this on the internet but was not able to find a working solution.

**Comment by vrisalab** [comment]
Hi,
I also need to enable the rtmp port to allow for live streaming on my peertube server, here the config I'm trying

> tunnel: <id>
credentials-file: /home/<user>/.cloudflared/<id>.json

ingress:
...
  - hostname: tube.server.net
    service: https://tube.server.net
    originServerName: tube.server.net
  - hostname: stream.tube.server.net
    service: rtmp://tube.server.net:1935
  - service: http_status:404

But this does not work, rtmpdump just times out as if the rule isn't even there. I don't get what I am doing wrong. I have tube.server.net set as localhost in my machine, so it should work since running

> rtmpdump -r rtmp://tube.server.net:1935/live/garbage

does work on my machine

**Comment by gvccvwangmingn2** [comment]
@vrisalab any errors from cloudflared client side? you cannot push rtmp directly to stream.tube.server.net

**Comment by vrisalab** [comment]
@gvccvwangmingn2 No errors on the cloudflared service logs. Is there any way to be able to do what I want?

**Comment by Milssss** [comment]
I'm trying to achieve this as well.. are we able to proxy a running RTMP server behind the Argo tunnel to our domain via subdomain?
I thought it was against ToS to run anything other than HTTP and HTTPS through.

**Comment by leetfin** [comment]
+1 to this issue, not able to a get a stream running.

**Comment by jorbig** [comment]
+1

Will the RTMP protocol ever be supported by Cloudflared? Or is there some kind of workaround?

**Comment by lvnilesh** [comment]
https://github.com/cloudflare/cloudflared/issues/14

**Comment by leetfin** [comment]
It has been four years, that response is no longer necessarily accurate.

**Comment by jorbig** [comment]
I agree with @leetfin. A protocol like RDP is supported as well, but that also doesn't run over HTTP, right?

**Comment by itsdoublearon** [comment]
Bump for 2024 - Any progress on this? Because it would be extremely helpful. Thanks.

**Comment by Scorpio-mfg** [comment]
any update on this ?

**Comment by a-h** [comment]
I got this working, but had to configure it in the dashboard, not via the CLI.

I went into CloudFlare One, then Networks, then Connectors. Then, I selected my tunnel, and went to the "Published application routes" tab, I could then add a TCP service pointing at localhost:31168 (I used a different domain for RTMP than web).

It's not obvious how to extract the config as code.

On the client that's pushing video: `cloudflared access tcp --hostname rtmp.$DOMAIN --url localhost:1935` then, I could push video to localhost:1935.
